ListableUI/Sources/ListableBuilder.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  ListableBuilder.swift
+//  ListableUI
+//
+//  Created by Kyle Van Essen on 6/10/21.
+//
+
+///
+/// A result builder which can be used to provide a SwiftUI-like DSL for building arrays of content.
+///
+/// You provide a result builder in an API by specifying it as a method parameter, like so:
+///
+/// ```
+/// init(@ListableBuilder<SomeContent> contents : () -> [SomeContent]) {
+///     self.contents = contents()
+/// }
+/// ```
+///
+/// ## Links & Videos
+/// https://github.com/apple/swift-evolution/blob/main/proposals/0289-result-builders.md
+/// https://developer.apple.com/videos/play/wwdc2021/10253/
+/// https://www.swiftbysundell.com/articles/deep-dive-into-swift-function-builders/
+/// https://www.avanderlee.com/swift/result-builders/
+///
+/// ### Note
+/// Most comments on methods come from the result builders SE proposal.
+///
+@resultBuilder public enum ListableBuilder<ContentType> {
+    
+    /// The type of individual statement expressions in the transformed function.
+    public typealias Expression = ContentType
+
+    /// The type of a partial result.
+    public typealias Component = [ContentType]
+
+    /// The type of the final returned result.
+    public typealias FinalResult = [ContentType]
+    
+    /// If an empty closure is provided, returns an empty array.
+    public static func buildBlock() -> Component {
+        []
+    }
+    
+    /// Required by every result builder to build combined results from statement blocks.
+    public static func buildBlock(_ components: Component...) -> Component {
+        components.reduce(into: []) { $0 += $1 }
```

Comment:
Can just do this - or is `into` more performant?
```suggestion
        components.reduce([], +)
```